src/weak_memory.rs

```diff
@@ -0,0 +1,271 @@
+use std::{
+    cell::{Ref, RefCell, RefMut},
+    collections::VecDeque,
+};
+
+use rustc_const_eval::interpret::{AllocRange, InterpResult, ScalarMaybeUninit};
+use rustc_data_structures::fx::FxHashMap;
+use rustc_target::abi::Size;
+
+use crate::{
+    data_race::{GlobalState, ThreadClockSet},
+    RangeMap, Tag, VClock, VTimestamp, VectorIdx,
+};
+pub type AllocExtra = StoreBufferAlloc;
```

Comment:
```suggestion

pub type AllocExtra = StoreBufferAlloc;
```
Keeping blocks of code (separated by empty lines) semantically meaningful can help quite a bit with readability.